homeassistant/components/somfy/cover.py

```diff
@@ -88,6 +138,17 @@ def is_closed(self):
             is_closed = self._closed
         return is_closed
 
+    @property
+    def assumed_state(self):
+        """Return if the cover has an assumed state."""
+        if not self.optimistic:
+            return None
+        return (
+            self._is_closing is not None
+            or self._is_opening is not None
+            or self._closed is not None
+        )
```

Comment:
Instead I suggest you to check if self.device.states is empty or undefined. That's the case for RTS devices.

Reply:
See this example: https://github.com/tetienne/somfy-open-api/issues/1

Reply:
Adjusted, retesting...

Reply:
Works as expected 👍